website_sale_secondary_unit/static/src/js/website_sale_secondary_unit.js

```diff
@@ -0,0 +1,102 @@
+odoo.define("website_sale_secondary_unit.animation", function(require) {
+    "use strict";
+
+    const sAnimation = require("website.content.snippets.animation");
+
+    sAnimation.registry.sale_secondary_unit = sAnimation.Class.extend({
+        selector: ".secondary-unit",
+        // eslint-disable-next-line no-unused-vars
+        init: function(parent, editableMode) {
+            this._super.apply(this, arguments);
+            this.$secondary_uom = null;
+            this.$secondary_uom_qty = null;
+            this.$product_qty = null;
+            this.secondary_uom_qty = null;
+            this.secondary_uom_factor = null;
+            this.product_uom_factor = null;
+            this.product_qty = null;
+        },
+        start: function() {
+            this.$secondary_uom = $("#secondary_uom");
+            this.$secondary_uom_qty = $(".secondary-quantity");
+            this.$product_qty = $(".quantity");
+            this._setValues();
+            this.$target.on(
+                "change",
+                ".secondary-quantity",
+                this._onChangeSecondaryUom.bind(this)
+            );
+            this.$target.on(
+                "change",
+                "#secondary_uom",
+                this._onChangeSecondaryUom.bind(this)
+            );
+            this.$product_qty.on("change", null, this._onChangeProductQty.bind(this));
+            if (this.secondary_uom_qty) {
+                this._onChangeSecondaryUom();
+            }
+        },
+        _setValues: function() {
+            this.secondary_uom_qty = parseFloat(
+                this.$target.find(".secondary-quantity").val()
+            );
+            this.secondary_uom_factor = parseFloat(
+                $("option:selected", this.$secondary_uom).data("secondary-uom-factor")
+            );
+            this.product_uom_factor = parseFloat(
+                $("option:selected", this.$secondary_uom).data("product-uom-factor")
+            );
+            this.product_qty = parseFloat($(".quantity").val());
```

Comment:
I think that here we want coercing the values instead of truncate the string value... no? For example. values like "1asas" should give an error not 1.
```suggestion
            this.secondary_uom_qty = Number(
                this.$target.find(".secondary-quantity").val()
            );
            this.secondary_uom_factor = Number(
                $("option:selected", this.$secondary_uom).data("secondary-uom-factor")
            );
            this.product_uom_factor = Number(
                $("option:selected", this.$secondary_uom).data("product-uom-factor")
            );
            this.product_qty = Number($(".quantity").val());
```